tests/python/pants_test/java/test_nailgun_io.py

```diff
@@ -76,3 +76,33 @@ def test_run_read_write(self, mock_writer, mock_select, mock_read):
       mock.call(mock.ANY, ChunkType.STDIN, b'A' * 300),
       mock.call(mock.ANY, ChunkType.STDIN_EOF)
     ])
+
+
+class TestPipedNailgunStreamWriter(unittest.TestCase):
+  def setUp(self):
+    self.mock_socket = mock.Mock()
+
+  @mock.patch('os.read')
+  @mock.patch('select.select')
+  @mock.patch.object(NailgunProtocol, 'write_chunk')
+  def test_auto_shutsdown_on_write_end_closed(self, mock_writer, mock_select, mock_read):
+    pipe = Pipe.create(False)
+    input = [b"A"] * 100000 + [b'']
```

Comment:
Maybe call this something other than `input`, as that's a keyword in the langauge?

Reply:
Done :)